File: games/hangman/hangman.py
import random
import constants

MAX_FAIL = 6

def welcome():
      print("###########################################")
      print("##  Bienvenidos al juego del Ahorcado    ##")
      print("###########################################")

def good_bye():
      print("###########################################")
      print("##     Adios, gracias por participar     ##")
      print("###########################################")

def get_random_word():
      index = random.randint(0,len(constants.WORDS)-1)
      return constants.WORDS[index]

def print_board(index, word, correct_letters):
      if (index <= MAX_FAIL and index >= 0 ):
            print(constants.IMAGES[index])
      
      for letter in word:
            if(letter in  correct_letters):
                  print(letter, end=' ')
            else:
                  print('_', end=' ')
      print()

def choose_letter():
      print('Elije una letra')

      while True:
            letter = input()
            if len(letter) == 1 and letter.isalpha():
                  return letter.lower()
            else:
                  print("Elije solo una letra")

def game():
      secrete_word = get_random_word()
      fail_counter = 0
      correct_letters = []

      print("**************")
      print(secrete_word)

      while True:
            print_board(fail_counter, secrete_word, correct_letters)
            letter = choose_letter()

            
            if(letter in secrete_word):
                  print('correcto adivinaste una letra')
                  if not (letter in correct_letters):
                        correct_letters.append(letter)
                  if(set(secrete_word) == set(correct_letters)):
                        print("Ganaste")
                        break
            else:
                  fail_counter = fail_counter + 1
                  print('fallaste')
                  print_board(fail_counter, secrete_word, correct_letters)

                  if (fail_counter == MAX_FAIL): 
                        print('has perdido la partida')
                        break

playAgain = "si"
welcome()

while playAgain == 'si' or playAgain == 's' or playAgain == 'SI' or playAgain == 'S':
      game()
      print("")
      print("¿Quieres jugar de nuevo? selecciona si o no")
      playAgain = input()

good_bye()

File: games/hangman/hangman.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

    const char IMG_0[] = 
"   +---+\n"
"   |   |\n"
"       |\n"
"       |\n"
"       |\n"
"       |\n"
"=========\n";
    const char IMG_1[] = 
"   +---+\n"
"   |   |\n"
"   O   |\n"
"       |\n"
"       |\n"
"       |\n"
"=========\n";

    const char IMG_2  [] = 
"   +---+\n"
"   |   |\n"
"   O   |\n"
"  /|   |\n"
"       |\n"
"       |\n"
"=========\n";

    const char IMG_3  [] = 
"   +---+\n"
"   |   |\n"
"   O   |\n"
"  /|\\  |\n"
"       |\n"
"       |\n"
"=========\n";

    const char IMG_4  [] = 
"   +---+\n"
"   |   |\n"
"   O   |\n"
"  /|\\  |\n"
"  /    |\n"
"       |\n"
"=========\n";

    const char IMG_5  [] = 
"   +---+\n"
"   |   |\n"
"   O   |\n"
"  /|\\  |\n"
"  / \\  |\n"
"       |\n"
"=========\n";
    
char *WORDS[6]  = {"limpiar", "cortar", "supercalifrajilistico", "espacios", "lagarto", "spiderman"};
const int MAX_FAIL = 6;


void welcome(){
    printf("###########################################\n");
    printf("##  Bienvenidos al juego del Ahorcado    ##\n");
    printf("###########################################\n");
    printf("\n");
}
 
void good_bye(){
    printf("###########################################\n");
    printf("##      Adios, gracias por participar    ##\n");
    printf("###########################################\n");
    printf("\n");
}     

int rand_number(){
    time_t t;
    int m = 0;
    int n = 7;
    srand((unsigned) time(&t));
    return rand () % (n-m+1) + m;
}
char* get_random_word(){
    int i = rand_number();
    return WORDS[i];
}

int is_letter_in_word(char letter, char *correct_letters, int size){
    for (int i = 0; i < size; i++){
        if (letter == correct_letters[i]){
           return 1;
        }
    }
    return 0;
}

void print_board(int index, char *word, char *correct_letters){

    if (index <= MAX_FAIL && index >= 0 ){
        
        switch (index){
            case 0: printf("%s", IMG_0);
                break;
            case 1: printf("%s", IMG_1);
                break;
            case 2: printf("%s", IMG_2);
                break;
            case 3: printf("%s", IMG_3);
                break;
            case 4: printf("%s", IMG_4);
                break;
            case 5: printf("%s", IMG_5);
                break;
        }
        
        int correct_letters_size = (int) sizeof(correct_letters);

        int word_size = (int) sizeof(word);
        for (int i = 0; i < word_size-1; i++){
            if (is_letter_in_word(word[i], correct_letters, correct_letters_size)){
                printf(" %c ", word[i]);
            }else{
                printf(" _ ");
            }
        }
    }
}


int main()
{
    welcome();
    good_bye();
    char *word;
    char correct_letters[] = {'a'};
    word = get_random_word();
    
    print_board(3, word, correct_letters);
    
    
    return 0;
}
